pkg/configuration/configuration.go

```diff
@@ -280,3 +291,76 @@ func (c *Config) KeyringPrefix() (string, error) {
 	}
 	return h, nil
 }
+
+func (c *Config) CheckForUpdate(out io.Writer, current string) (Meta, error) {
+	// Check if version check is disabled in configuration
+	if c.Meta.DisableVersionCheck {
+		return c.Meta, errors.New("version check disabled")
+	}
+
+	// Check if version check has already been done today
+	yesterday := time.Now().AddDate(0, 0, -1)
+	lastCheck, err := time.Parse(time.RFC3339Nano, c.Meta.LastChecked)
+	if err != nil {
+		return c.Meta, err
+	}
+	if !lastCheck.Before(yesterday) {
+		return c.Meta, errors.New("version check already done today")
+	}
```

Comment:
```suggestion
	// Check if version check has already been done today
	lastCheck, err := time.Parse(time.RFC3339Nano, c.Meta.LastChecked)
	if err == nil {
        yesterday := time.Now().AddDate(0, 0, -1)
		if !lastCheck.Before(yesterday) {
			return c.Meta, errors.New("version check already done today")
		}
	}
```

otherwise it will always fail here if configure through env variables without any config file